stock_serial_batch_receive/wizard/stock_move_line_serial_generator.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError, UserError
+
+
+class StockMoveLineSerialGenerator(models.TransientModel):
+
+    _name = 'stock.move.line.serial.generator'
+    _description = 'Auto generate serial numbers according to product sequence'
+
+    stock_move_id = fields.Many2one('stock.move', readonly=True)
+    product_id = fields.Many2one(
+        'product.product', related='stock_move_id.product_id'
+    )
+    picking_type_create_lots = fields.Boolean(
+        related='stock_move_id.picking_type_create_lots'
+    )
+    qty_to_process = fields.Integer(
+        help="Defines how many serial numbers will be generated on stock move "
+        "lines without a serial number yet."
+    )
+    first_number = fields.Char(copy=False)
```

Comment:
```suggestion
    first_number = fields.Char()
```
It's a wizard so this arg shouldn't be needed.